Description: Patch sage-env and env.py
 Note that the patched sage-env is only used during the package build and not installed.
 Instead the file debian/adhoc/sage-env is installed.
 .
 The LD_LIBRARY_PATH can probably be forwarded upstream, either way is fine.
Author: Tobias Hansen <thansen@debian.org>, Ximin Luo <infinity0@debian.org>

--- a/sage/src/bin/sage
+++ b/sage/src/bin/sage
@@ -225,6 +225,23 @@
     exit $status
 fi
 
+#####################################################################
+# Source sage-env ($0 is the name of this "sage" script, so we can just
+# append -env to that). We redirect stdout to stderr, which is safer
+# for scripts.
+#####################################################################
+if [ "\"$0\"" = "\"/usr/bin/sage\"" ]; then
+    . "/usr/share/sagemath/bin/sage-env" >&2
+else
+    . "$0-env" >&2
+fi
+
+if [ $? -ne 0 ]; then
+    echo >&2 "Error setting environment variables by sourcing '$0-env';"
+    echo >&2 "possibly contact sage-devel (see http://groups.google.com/group/sage-devel)."
+    exit 1
+fi
+
 
 #####################################################################
 # Report information about the Sage environment
@@ -255,19 +272,7 @@
 fi
 
 
-#####################################################################
-# Source sage-env ($0 is the name of this "sage" script, so we can just
-# append -env to that). We redirect stdout to stderr, which is safer
-# for scripts.
-#####################################################################
-if [ -f "$0-env" ]; then
-    . "$0-env" >&2
-    if [ $? -ne 0 ]; then
-        echo >&2 "Error setting environment variables by sourcing '$0-env';"
-        echo >&2 "possibly contact sage-devel (see http://groups.google.com/group/sage-devel)."
-        exit 1
-    fi
-fi
+
 
 if [ -z "$DOT_SAGE" ]; then
     export DOT_SAGE="$HOME/.sage"
--- a/sage/src/bin/sage-env
+++ b/sage/src/bin/sage-env
@@ -283,13 +283,14 @@
 # depending on SAGE_ROOT and SAGE_LOCAL which are already defined.
 export SAGE_ETC="$SAGE_LOCAL/etc"
 export SAGE_SHARE="$SAGE_LOCAL/share"
-export SAGE_EXTCODE="$SAGE_SHARE/sage/ext"
+export SAGE_EXTCODE="$SAGE_SHARE/sagemath/ext"
 export SAGE_SPKG_INST="$SAGE_ROOT/local/var/lib/sage/installed"
 export SAGE_SPKG_SCRIPTS="$SAGE_LOCAL/var/lib/sage/scripts"
 export SAGE_LOGS="$SAGE_ROOT/logs/pkgs"
 export SAGE_SRC="$SAGE_ROOT/src"
 export SAGE_DOC_SRC="$SAGE_SRC/doc"
-export SAGE_DOC="$SAGE_SHARE/doc/sage"
+export SAGE_DOC="$SAGE_SHARE/doc/sagemath"
+export SAGE_ARB_LIBRARY=flint-arb
 
 if [ -z "${SAGE_ORIG_PATH_SET}" ]; then
     SAGE_ORIG_PATH=$PATH && export SAGE_ORIG_PATH
@@ -342,8 +343,8 @@
 [ -z "$CPATH" ] || CPATH="${CPATH}:"
 export CPATH="${CPATH}$SAGE_LOCAL/include"
 
-SINGULARPATH="$SAGE_LOCAL/share/singular" && export SINGULARPATH
-SINGULAR_EXECUTABLE="$SAGE_LOCAL/bin/Singular" && export SINGULAR_EXECUTABLE
+SINGULARPATH="/usr/share/singular" && export SINGULARPATH
+SINGULAR_EXECUTABLE="/usr/bin/Singular" && export SINGULAR_EXECUTABLE
 
 if [ -z "$SAGE_REPO_ANONYMOUS" ]; then
     SAGE_REPO_ANONYMOUS="git://trac.sagemath.org/sage.git"
@@ -358,16 +359,6 @@
     export SAGE_DISTFILES
 fi
 
-# Check that $HOME exists
-if [ "$HOME" = "" ]; then
-    echo >&2 'Error: environment variable $HOME is not set.'
-    return 1
-fi
-if ! [ -d "$HOME" ]; then
-    echo >&2 "Error: HOME directory '$HOME' does not exist."
-    return 1
-fi
-
 if [ "$DOT_SAGE" = "" ]; then
     # It is *not* an error if this directory does not exist, it will
     # be created in src/bin/sage or src/sage/misc/misc.py.
@@ -410,8 +401,10 @@
     unset PYTHONHOME
 fi
 
-LDFLAGS="-L$SAGE_LOCAL/lib -Wl,-rpath,$SAGE_LOCAL/lib $LDFLAGS"
+LDFLAGS="-L$SAGE_LOCAL/lib $LDFLAGS"
 export LDFLAGS
+LD_LIBRARY_PATH="$SAGE_LOCAL/lib:$LD_LIBRARY_PATH"
+export LD_LIBRARY_PATH
 
 if [ -z "$IPYTHONDIR" ]; then
     # We hardcode a version number in the directory name. The idea is
@@ -473,7 +466,6 @@
     fi
 fi
 
-export MAXIMA_PREFIX="$SAGE_LOCAL"
 export MAXIMA_USERDIR="$DOT_SAGE/maxima"
 
 PERL5LIB="$SAGE_LOCAL/lib/perl5:$PERL5LIB" && export PERL5LIB
@@ -585,9 +577,6 @@
     LD_LIBRARY_PATH="$SAGE_LOCAL/bin:$LD_LIBRARY_PATH" && export LD_LIBRARY_PATH
 fi
 
-# See trac 7186 -- this is needed if ecl is moved
-ECLDIR="$SAGE_LOCAL/lib/ecl/" && export ECLDIR
-
 # Handle parallel building/testing/...
 # See Trac Ticket #12016
 # First, figure out the right values for SAGE_NUM_THREADS (default
--- a/sage/build/make/install
+++ b/sage/build/make/install
@@ -16,7 +16,7 @@
 fi
 
 export SAGE_SHARE="$SAGE_LOCAL/share"
-export SAGE_EXTCODE="$SAGE_SHARE/sage/ext"
+export SAGE_EXTCODE="$SAGE_SHARE/sagemath/ext"
 export SAGE_LOGS="$SAGE_ROOT/logs/pkgs"
 export SAGE_SPKG_INST="$SAGE_ROOT/local/var/lib/sage/installed"
 . "$SAGE_SRC"/bin/sage-version.sh
--- a/sage/build/make/deps
+++ b/sage/build/make/deps
@@ -249,7 +249,7 @@
 	cd "$(SAGE_SRC)/doc" && $(MAKE) clean
 
 doc-output-clean:
-	rm -rf "$(SAGE_SHARE)/doc/sage"
+	rm -rf "$(SAGE_SHARE)/doc/sagemath"
 
 
 ###############################################################################
--- a/sage/src/doc/en/developer/sage_manuals.rst
+++ b/sage/src/doc/en/developer/sage_manuals.rst
@@ -54,7 +54,7 @@
 
     sage --docbuild tutorial html
 
-You can now open ``SAGE_ROOT/local/share/doc/sage/html/en/tutorial/index.html`` in
+You can now open ``SAGE_LOCAL/share/doc/sagemath/html/en/tutorial/index.html`` in
 your web browser.
 
 - Do you want to **add a new file** to the documentation? :ref:`Click here
--- a/sage/src/doc/en/faq/faq-general.rst
+++ b/sage/src/doc/en/faq/faq-general.rst
@@ -316,7 +316,7 @@
 documentation that makes up the Sage standard documentation. If you
 have downloaded a binary Sage release, the HTML version of the
 corresponding documentation comes pre-built and can be found under the
-directory ``SAGE_ROOT/local/share/doc/sage/html/``.
+directory ``SAGE_LOCAL/share/doc/sagemath/html/``.
 During the compilation of Sage from source, the HTML version of the
 documentation is also built in the process. To build the HTML version
 of the documentation, issue the following command from ``SAGE_ROOT``:
--- a/sage/src/doc/en/installation/source.rst
+++ b/sage/src/doc/en/installation/source.rst
@@ -886,7 +886,7 @@
   primarily intended for use when producing certain binary
   distributions of Sage, to lower the size of the distribution. As of
   this writing (December 2014, Sage 6.5), there are only a few such
-  plots, adding about 4M to the :file:`local/share/doc/sage/` directory.
+  plots, adding about 4M to the :file:`/usr/share/doc/sagemath/` directory.
   In the future, this may grow, of course. Note: after using this, if you
   want to build the documentation and include the pictures, you should
   run ``make doc-clean``, because the presence, or lack, of pictures
--- a/sage/src/sage_setup/docbuild/__init__.py
+++ b/sage/src/sage_setup/docbuild/__init__.py
@@ -415,7 +415,7 @@
             document_name = document.split('/')[1]
 
             # the sage directory within a subdocument, for example
-            # local/share/doc/sage/html/en/reference/algebras/sage
+            # $SAGE_LOCAL/share/doc/sagemath/html/en/reference/algebras/sage
             sage_directory = os.path.join(path, document, 'sage')
 
             # Walk through all of the files in the sage_directory
--- a/sage/src/sage/env.py
+++ b/sage/src/sage/env.py
@@ -4,15 +4,6 @@
 AUTHORS:
 
 - \R. Andrew Ohana (2012): Initial version.
-
-Verify that Sage can be started without any ``SAGE_`` environment
-variables::
-
-    sage: env = {k:v for (k,v) in os.environ.items() if not k.startswith("SAGE_")}
-    sage: import subprocess
-    sage: cmd = "from sage.all import SAGE_ROOT; print(SAGE_ROOT)"
-    sage: res = subprocess.call([sys.executable, "-c", cmd], env=env)  # long time
-    None
 """
 
 # ****************************************************************************
@@ -153,7 +144,7 @@
 # bunch of sage directories and files
 var('SAGE_LOCAL',          os.path.abspath(sys.prefix))
 var('SAGE_ETC',            join(SAGE_LOCAL, 'etc'))
-var('SAGE_SCRIPTS_DIR',    join(SAGE_LOCAL, 'bin'))
+var('SAGE_SCRIPTS_DIR',    '/usr/share/sagemath/bin')
 var('SAGE_INC',            join(SAGE_LOCAL, 'include'))
 var('SAGE_SHARE',          join(SAGE_LOCAL, 'share'))
 var('SAGE_DOC',            join(SAGE_SHARE, 'doc', 'sage'))
@@ -171,23 +162,23 @@
 
 # installation directories for various packages
 var('SAGE_EXTCODE',                  join(SAGE_SHARE, 'sage', 'ext'))
-var('CONWAY_POLYNOMIALS_DATA_DIR',   join(SAGE_SHARE, 'conway_polynomials'))
-var('GRAPHS_DATA_DIR',               join(SAGE_SHARE, 'graphs'))
-var('ELLCURVE_DATA_DIR',             join(SAGE_SHARE, 'ellcurves'))
-var('POLYTOPE_DATA_DIR',             join(SAGE_SHARE, 'reflexive_polytopes'))
-var('GAP_ROOT_DIR',                  join(SAGE_SHARE, 'gap'))
-var('THEBE_DIR',                     join(SAGE_SHARE, 'thebe'))
-var('COMBINATORIAL_DESIGN_DATA_DIR', join(SAGE_SHARE, 'combinatorial_designs'))
-var('CREMONA_MINI_DATA_DIR',         join(SAGE_SHARE, 'cremona'))
+var('CONWAY_POLYNOMIALS_DATA_DIR',   '/usr/share/sagemath/conway_polynomials')
+var('GRAPHS_DATA_DIR',               '/usr/share/sagemath/graphs')
+var('ELLCURVE_DATA_DIR',             '/usr/share/sagemath/ellcurves')
+var('POLYTOPE_DATA_DIR',             '/usr/share/sagemath/reflexive_polytopes')
+var('GAP_ROOT_DIR',                  '/usr/share/gap')
+var('THEBE_DIR',                     '/usr/share/thebe')
+var('COMBINATORIAL_DESIGN_DATA_DIR', '/usr/share/sagemath/combinatorial_designs')
+var('CREMONA_MINI_DATA_DIR',         '/usr/share/sagemath/cremona')
 var('CREMONA_LARGE_DATA_DIR',        join(SAGE_SHARE, 'cremona'))
-var('JMOL_DIR',                      join(SAGE_SHARE, 'jmol'))
+var('JMOL_DIR',                      '/usr/share/jmol')
 var('JSMOL_DIR',                     join(SAGE_SHARE, 'jsmol'))
-var('MATHJAX_DIR',                   join(SAGE_SHARE, 'mathjax'))
+var('MATHJAX_DIR',                   '/usr/share/javascript/mathjax')
 var('MTXLIB',                        join(SAGE_SHARE, 'meataxe'))
-var('THREEJS_DIR',                   join(SAGE_SHARE, 'threejs'))
+var('THREEJS_DIR',                   '/usr/share/javascript/three')
 var('SINGULARPATH',                  join(SAGE_SHARE, 'singular'))
-var('PPLPY_DOCS',                    join(SAGE_SHARE, 'doc', 'pplpy'))
-var('MAXIMA_FAS')
+var('PPLPY_DOCS',                    '/usr/share/doc/python-ppl-doc/html')
+var('MAXIMA_FAS',                    '/usr/lib/ecl/maxima.fas')
 
 # misc
 var('SAGE_BANNER', '')
